FurCoNZ/ViewComponents/StripeViewComponent.cs

```diff
@@ -0,0 +1,126 @@
+        private readonly IOptions<StripeSettings> _stripeOptions;
+
+        public StripeViewComponent(IUserService userService, IOrderService orderService, StripeService stripeService, IOptions<StripeSettings> stripeOptions)
+        {
+            _userService = userService;
+            _orderService = orderService;
+            _stripeService = stripeService;
+            _stripeOptions = stripeOptions;
+        }
+
+        public async Task<IViewComponentResult> InvokeAsync(int orderId)
+        {
+            var user = await _userService.GetCurrentUserAsync(HttpContext.RequestAborted);
+            if (user == null)
+                return View(); // Default view displays nothing to the end user.
+
+            var order = await _orderService.GetUserOrderAsync(user, orderId, HttpContext.RequestAborted);
+            if (order == null)
+                return View(); // Default view displays nothing to the end user.
+
+            if (order.AmountPaidCents >= order.TotalAmountCents)
+                return View(); // Default view displays nothing to the end user.
+
+            var lineItems = order.TicketsPurchased.GroupBy(t => t.TicketType, new TicketTypeComparer());
+
+            Stripe.StripeConfiguration.SetApiKey(_stripeOptions.Value.SecretKey);
+
+            var checkoutSessionOptions = new Stripe.Checkout.SessionCreateOptions
+            {
+                // TODO: Make PaymentMethodTypes configurable
+                PaymentMethodTypes = new List<string> { "card" },
+                CustomerEmail = user.Email,
+                LineItems = lineItems.Select(l => new Stripe.Checkout.SessionLineItemOptions {
+                    Name = l.Key.Name,
+                    Amount = l.Key.PriceCents,
+                    Description = l.Key.Description,
+                    Currency = "NZD",
+                    Quantity = l.Count(), // This need to be calculated
+                }).ToList(),
+                ClientReferenceId = $"order_{orderId}",
+                PaymentIntentData = new Stripe.Checkout.SessionPaymentIntentDataOptions
+                {
+                    // Pass in metadata that can be easily viewed from Stripe Dashboard
+                    Metadata = new Dictionary<string, string>
+                    {
+                        {"client_id", user.Id.ToString() },
+                        {"client_name", user.Name },
+                        {"client_email", user.Email },
+                        {"order_id", order.Id.ToString() },
+                    },
+                },
+                SuccessUrl = Url.Action("Success", "Stripe", null, Request.Scheme),
+                CancelUrl = Url.Action("Cancelled", "Stripe", null, Request.Scheme),
+            };
+
+            var subTotal = checkoutSessionOptions.LineItems.Sum(l => l.Amount.Value);
+            // Calculate stripe fee to ensure received amount is what's requested.
+            var stripeFee = ((subTotal + 30) * 1000 / 971) - subTotal;
```

Comment:
Will be good to eventually pull this through via `IOptions<StripeSettings>`, since these values can change and probably shouldn't be hardcoded